cli/flox/src/commands/activate.rs

```diff
@@ -551,16 +551,64 @@ impl Activate {
                 Shell::detect_from_env("SHELL")
             })
     }
+
+    /// Construct the envrionment list for the shell prompt
+    ///
+    /// [`None`] if the prompt is disabled, or filters removed all components.
+    fn make_environment_prompt(
+        prompt_display_config: EnvironmentPromptConfig,
+        flox_active_environments: &super::ActiveEnvironments,
+    ) -> Option<String> {
+        if prompt_display_config == EnvironmentPromptConfig::HideAll {
+            return None;
+        }
+
+        let prompt_envs: Vec<_> = flox_active_environments
+            .iter()
+            .filter_map(|env| {
+                if prompt_display_config == EnvironmentPromptConfig::HideDefault
+                    && env.name().as_ref() == DEFAULT_NAME
+                {
+                    return None;
+                }
+                Some(env.bare_description().expect("None of the branches panics"))
```

Comment:
nit: not really sure what `None of the branches panics` means

Reply:
`bare_description()` returns a result but never returns Err, i can change the message to be a bit more clear